src/rulesets/LinearAlgebra/dense.jl

```diff
@@ -11,15 +11,9 @@ function rrule(::typeof(dot), x::AbstractArray, y::AbstractArray)
     project_y = ProjectTo(y)
     function dot_pullback(Ω̄)
         ΔΩ = unthunk(Ω̄)
-        xthunk = InplaceableThunk(
-            dx -> dx .+= reshape(y, axes(x)) .* ΔΩ',
-            @thunk(project_x(reshape(y .* ΔΩ', axes(x)))),
-        )
-        ythunk = InplaceableThunk(
-            dy -> dy .+= reshape(x, axes(y)) .* ΔΩ,
-            @thunk(project_y(reshape(x .* ΔΩ, axes(y)))),
-        )
-        return (NoTangent(), xthunk, ythunk)
+        x̄ = @thunk(project_x(reshape(y .* ΔΩ', axes(x))))
```

Comment:
Is this permanent or until we have something like https://github.com/JuliaDiff/ChainRulesCore.jl/pull/393?

Reply:
Yeah,
more generally I hate the InplacableThunks that are just deferring broadcast materialization.
They are so boilerplatey.
That has its own issue https://github.com/JuliaDiff/ChainRulesCore.jl/issues/274
and is probably going to be easier to solve.

I am more than happy to delete them.
Can always add them back